packages/admin/cms-admin/src/generator/future/utils/generateFieldList.ts

```diff
@@ -0,0 +1,43 @@
+import { IntrospectionField, IntrospectionQuery, IntrospectionType } from "graphql";
+import objectPath from "object-path";
+
+export function generateFieldListGqlString(fields: string[]) {
+    const fieldsObject = fields.reduce<object>((acc, fieldName) => {
+        objectPath.set(acc, fieldName, true);
+        return acc;
+    }, {});
+    return JSON.stringify(fieldsObject, null, 4)
+        .replace(/: true,/g, "")
+        .replace(/: true/g, "")
+        .replace(/"/g, "")
+        .replace(/:/g, "");
```

Comment:
Hm, you are messing with the JSON output string, wouldn't a recursive function be relatively easy to implement?

Reply:
that's right. but I wanted to return nice indentations and json is somewhat similar. I didn't check, but is the code-format done with some prettier command-call? As I saw no "ugly" code. So indentations are probably not even needed, and a flat string would be much easier generated so I could change this json-messing.

Reply:
done. changed it to recursive function